sdk/storage/storage-blob/review/generated/src/models/index.d.ts

```diff
@@ -0,0 +1,6713 @@
+/// <reference types="node" />
+import { ServiceClientOptions } from "@azure/core-http";
+import * as coreHttp from "@azure/core-http";
+/**
+ * Key information
+ */
+export interface KeyInfo {
+    /**
+     * The date-time the key is active in ISO 8601 UTC time
+     */
+    start: string;
+    /**
+     * The date-time the key expires in ISO 8601 UTC time
+     */
+    expiry: string;
+}
+/**
+ * A user delegation key
+ */
+export interface UserDelegationKey {
+    /**
+     * The Azure Active Directory object ID in GUID format.
+     */
+    signedOid: string;
+    /**
+     * The Azure Active Directory tenant ID in GUID format
+     */
+    signedTid: string;
+    /**
+     * The date-time the key is active
+     * **NOTE: This entity will be treated as a string instead of a Date because the API can
+     * potentially deal with a higher precision value than what is supported by JavaScript.**
+     */
+    signedStart: string;
+    /**
+     * The date-time the key expires
+     * **NOTE: This entity will be treated as a string instead of a Date because the API can
+     * potentially deal with a higher precision value than what is supported by JavaScript.**
+     */
+    signedExpiry: string;
+    /**
+     * Abbreviation of the Azure Storage service that accepts the key
+     */
+    signedService: string;
+    /**
+     * The service version that created the key
+     */
+    signedVersion: string;
+    /**
+     * The key as a base64 string
+     */
+    value: string;
+}
+/**
+ * An interface representing StorageError.
+ */
+export interface StorageError {
+    message?: string;
+}
+/**
+ * The service error response object.
+ */
+export interface DataLakeStorageErrorError {
```

Comment:
Should there be a type in azure-core that represents error code + error message?

Reply:
There is, though extending doesn't provide much value unless there's an uber-error type that people will want to catch.